src/shapley_sensitivity.jl

```diff
@@ -105,36 +105,35 @@ function find_cond_mean_var(cov::Matrix,
     Find the conditional mean and variance of the given distribution
     """
 
-    mat_b = cov[:, dependent_ind]
-    mat_b = mat_b[dependent_ind, :]
+    mat_b = cov[dependent_ind, dependent_ind]
 
-    mat_c = cov[:, dependent_ind]
-    mat_c = mat_c[given_ind, :]
+    mat_c = cov[given_ind, dependent_ind]
 
-    mat_d = cov[:, given_ind]
-    mat_d = mat_d[given_ind, :]
+    mat_d = cov[given_ind, given_ind]
 
-    mat_cdinv = transpose(mat_c) * inv(mat_d)
+    mat_cdinv = transpose(mat_c) / mat_d
     conditional_mean = mat_cdinv * X_given
     conditional_var = mat_b - mat_cdinv * mat_c
 
     return conditional_mean, conditional_var
 end
 
-function cond_sampling(distribution::SklarDist{IndependentCopula{d},Tm},
+function cond_sampling(distribution::SklarDist{<:IndependentCopula},
     n_sample::Int,
     idx::Vector{Int},
     idx_c::Vector{Int},
-    x_cond::AbstractArray) where {d,Tm}
+    x_cond::AbstractArray)
 	# conditional sampling in independent random vector is just subset sampling.
-    Array(rand(Copulas.subsetdims(distribution, idx), n_sample)') # this might need to be transposed. 
+    samples = zeros(eltype(x_cond), length(idx), n_sample)
```

Comment:
Preallocation is a bit safer, otherwise the `eltype` of the samples can be a bit surprising (see below).